src/io/iter_normalize.h

```diff
@@ -144,35 +148,71 @@ class ImageNormalizeIter : public IIterator<DataInst> {
     float illumination =
         rand_uniform(rnd_) * param_.max_random_illumination * 2 - param_.max_random_illumination;
 
-    if (param_.mean_r > 0.0f || param_.mean_g > 0.0f ||
-        param_.mean_b > 0.0f || param_.mean_a > 0.0f) {
-      // subtract mean per channel
-      data[0] -= param_.mean_r;
-      if (data.shape_[0] >= 3) {
-        data[1] -= param_.mean_g;
-        data[2] -= param_.mean_b;
-      }
-      if (data.shape_[0] == 4) {
-        data[3] -= param_.mean_a;
+    // setup mean if not loaded from file, should happen only once
+    if (!meanfile_ready_ || (meanimg_.shape_ != data.shape_)) {
+      if (param_.mean_r > 0.0f || param_.mean_g > 0.0f ||
+          param_.mean_b > 0.0f || param_.mean_a > 0.0f) {
+        // setup meanimg using mean_r/g/b/a
+        meanimg_.Resize(data.shape_);
+        switch (static_cast<int>(data.shape_[0])) {
+          case 4: {
+            meanimg_[3] = param_.mean_a;
```

Comment:
why modify meanimg and stdimg? Feels like a bad idea

Reply:
To condense operation and avoid too many branches for each input.
the meanimg_ and stdimg_ should be modified only once for the first image. I would like it to happen in Init, but can't infer the shape there.

Reply:
But reshaping and re filling meanimg on each iter is slow